doc/nrf/ug_matter_creating_accessory.rst

```diff
@@ -250,76 +255,68 @@ To add a new event handler, complete the following steps:
               }
       }
 
-Include headers for clusters' callbacks
-=======================================
+Include header for cluster attribute helpers
+============================================
 
-To have the required callback declaration for the Temperature Measurement cluster, make sure to include the following file in the :file:`src/app_task.cpp` file:
+To import helper functions for accessing cluster attributes, make sure to include the following file in the :file:`src/app_task.cpp` file:
 
 .. code-block:: C++
 
-   #include <app/clusters/temperature-measurement-server/temperature-measurement-server.h>
+   #include <app-common/zap-generated/attributes/Accessors.h>
 
 .. rst-class:: numbered-step
 
 Create a callback for sensor activation and deactivation
 ********************************************************
 
-Up until now the events :c:struct:`SensorActivate` and :c:struct:`SensorDeactivate` are not present in the event queue.
-This is because the attribute of the On/off cluster is not changed from within the ``AppTask``.
-It can only be changed externally, for example using the Matter controller.
-
-However, the sensor application needs to know how this attribute is changing to know when the sensor is turned on and off.
-This means that there is a need to implement a callback function to post one of these events every time the On/off attribute changes.
+Up until now handlers for the :c:struct:`SensorActivate` and :c:struct:`SensorDeactivate` events are ready, but the events are never posted to the event queue.
+The sensor is supposed to be turned on and off remotely by changing the OnOff attribute of the On/off cluster, for example using the Matter controller.
+This means that there is a need to implement a callback function to post one of these events every time the OnOff attribute changes.
 
 To implement the callback function, create a new file, for example :file:`src/zcl_callbacks.cpp`, and implement the callback in this file.
-To see a list of callback functions that are customizable, open :file:`src/gen/callback-stub.cpp` and look for functions with ``__attribute__((weak))``.
-It is sufficient just to implement ``emberAfPostAttributeChangeCallback`` (read the description in :file:`src/gen/callback-stub.cpp`).
+To see a list of callback functions that are customizable, open :file:`src/zap-generated/callback-stub.cpp` and look for functions with ``__attribute__((weak))``.
+It is sufficient just to implement ``MatterPostAttributeChangeCallback`` (read the description in :file:`src/zap-generated/callback-stub.cpp`).
```

Comment:
```suggestion
To implement the callback function, complete the following steps:

1. Create a new file, for example :file:`src/zcl_callbacks.cpp`.
2. Implement the callback in this file:

   a. Open :file:`src/zap-generated/callback-stub.cpp` to check the list of callback functions that are customizable.
   #. In the :file:`src/zap-generated/callback-stub.cpp` file, look for functions with ``__attribute__((weak))``.
   #. Read the description of ``MatterPostAttributeChangeCallback``.
   #. Implement ``MatterPostAttributeChangeCallback`` in the :file:`src/zcl_callbacks.cpp` file.
```
Would this work?

Reply:
Yes, but I merged points 'a' and 'b' as `__attribute__((weak))` actually indicates that the function is customizable.